ocaml/xapi/workload_balancing.ml

```diff
@@ -675,7 +675,7 @@ let get_evacuation_recoms ~__context ~uuid =
       []
     else
       match inner_xml with
-      | (Xml.Element ( _, _, [Xml.Element("CanPlaceAllVMs",_,_)])) ->
+      | (Xml.Element ( _, _, [_;_;Xml.Element("CanPlaceAllVMs",_,_)])) ->
```

Comment:
@liulinC commented on the main pull-request to explain that this is to match xml such as
`<HostGetRecommendationsResult i="http://www.w3.org/2001/XMLSchema-instance"> <ErrorMessage nil="true"/> <ResultCode>0</ResultCode> <CanPlaceAllVMs>true</CanPlaceAllVMs> </HostGetRecommendationsResult>`
but this code is fragile: it assumes that `HostGetRecommendationsResult` will have exactly three children, and that `CanPlaceAllVMs` will be the third of them.

The order of those three elements is essentially arbitrary, yes?
So I would prefer the code to work regardless of the order of the elements.
Perhaps something like
```
| (Xml.Element ( _, _, children)) when List.exists
    (function
       | Xml.Element("CanPlaceAllVMs",_,_) -> true
       | _ -> false
    ) children ->
  []
```